haec-user-service/src/_middleware.ts

```diff
@@ -0,0 +1,19 @@
+import { Injectable, NestMiddleware, Request, Response } from '@nestjs/common';
+import { NextFunction } from 'express';
+
+@Injectable()
+export class AuthenticationMiddleware implements NestMiddleware {
+
+  use(req: Request, res: Response, next: NextFunction) {
+    console.log("Middleware Authentication is running");
+    // Middleware Authentication SUCCESS
+    if (req.headers["haec-auth-token"] === "neither-is-this-a-secret-key") {
```

Comment:
I plan to update this line once this is pushed to the AWS server; is there something I should do differently now?

Reply:
This should be an environment variable just like what you have in the Django service, but for the NestJS service. 

If you could provide a dummy value [here in the NestJS `.env.example` file](https://github.com/mstephenhunt/holiday_acres/blob/main/haec-user-service/.env.example), you can simply reference environment variables with a line like:

```ts
env("MY_ENV_VAR_NAME")
```

Reply:
That makes total sense. On it.